client/src/Components/AppContext.jsx

```diff
@@ -0,0 +1,4 @@
+import { createContext } from 'react';
+
+const AppContext = createContext();
+export default AppContext;
```

Comment:
I'm sad you aren't using a custom Provider and Hook for this. Wouldn't it be nice in your client code to just call `useDisplayType()`? If you did that, you could even move the `useMediaQuery` call into here and hide even that from your code.

This is a way to make your code look more impressive. It's not as important as getting the code working, though.